core/lib/server/pub.ts

```diff
@@ -1398,6 +1480,113 @@ export async function getPubsWithRelatedValuesAndChildren<
 						)
 					)
 			)
+			.with("stage_ms", (db) =>
+				db
+					.selectFrom("stage_memberships")
+					.$if(Boolean(props.userId), (qb) =>
+						qb
+							.where("stage_memberships.userId", "=", props.userId!)
+							.select([
+								"role",
+								"stageId as membId",
+								sql<MembershipType>`'stage'::"MembershipType"`.as("type"),
+							])
+					)
+					.$castTo<{
+						role: MembershipCapabilitiesRole;
+						membId: string;
+						type: MembershipType;
+					}>()
+			)
+			.with("pub_ms", (db) =>
+				db
+					.selectFrom("pub_memberships")
+					.$if(Boolean(props.userId), (qb) =>
+						qb
+							.where("pub_memberships.userId", "=", props.userId!)
+							.select([
+								"role",
+								"pubId as membId",
+								sql<MembershipType>`'pub'::"MembershipType"`.as("type"),
+							])
+					)
+					.$castTo<{
+						role: MembershipCapabilitiesRole;
+						membId: string;
+						type: MembershipType;
+					}>()
+			)
+			.with("community_ms", (db) =>
+				db
+					.selectFrom("community_memberships")
+					.$if(Boolean(props.userId), (qb) =>
+						qb
+							.where("community_memberships.userId", "=", props.userId!)
+							.where("community_memberships.communityId", "=", props.communityId)
+							.select([
+								"role",
+								"communityId as membId",
+								sql<MembershipType>`'community'::"MembershipType"`.as("type"),
+							])
+					)
+					// Add a fake community admin role when selecting without a userId
```

Comment:
It's pretty hard to conditionally add a CTE so I ended up with all these $if statements to handle when a userId isn't passed. And looking at this now, I realize it would probably improve performance to use a $if on the `where` condition that actually enforces the authorization check instead of inserting this dummy row.